bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsStrategy.scala

```diff
@@ -85,12 +85,30 @@ object AgsStrategy {
      * Creates a new TargetEnvironment with guide stars for each assignment in
      * the Selection.
      */
-    def applyTo(env: TargetEnvironment): TargetEnvironment =
+    def applyTo(env: TargetEnvironment): TargetEnvironment = {
+      def findMatching(gpt: GuideProbeTargets, target: SPTarget): Option[SPTarget] =
+        Option(target.getTarget.getName).flatMap { n =>
+          gpt.getTargets.asScalaList.find { t =>
+            Option(t.getTarget.getName).map(_.trim).exists(_ == n.trim)
```

Comment:
Good call. I just pulled this out of the old pre-BAGS code and copied it in verbatim, but I will make the recommended change.